gnuradio-runtime/lib/CMakeLists.txt

```diff
@@ -142,9 +142,37 @@ if(ENABLE_CTRLPORT_THRIFT)
 FIND_PACKAGE(THRIFT)
 
 if(THRIFT_FOUND)
+
 list(APPEND EXTRA_DEPS "THRIFT")
 MATH(EXPR CTRLPORT_BACKENDS "${CTRLPORT_BACKENDS} + 1")
 
+# Thrift was found! Make sure the selected c++ standard is
+# compatible: Thrift <= 0.10 requires using boost:shared_ptr, which
+# will work with any c++ standard. Thrift >= 0.11.0 requires using
+# std::shared_ptr, which requires c++11 or later.
+
+if(${THRIFT_VERSION} VERSION_GREATER_EQUAL 0.11.0)
+  if(CMAKE_CXX_STANDARD)
+    if(${CMAKE_CXX_STANDARD} VERSION_LESS 11)
+      message(WARNING "\nWarning: Thrift version is ${THRIFT_VERSION}, which requires C++11 for building. The current c++ standard to use is C++${CMAKE_CXX_STANDARD}, which will not work with this Thrift version. Trying to set CMake internally to use C++11 ...")
+      set(CMAKE_CXX_STANDARD 11)
+    endif()
+  else()
+    message(WARNING "\nWarning: Thrift version is ${THRIFT_VERSION}, which requires C++11 for building. The c++ standard to use is not currently set. Trying to set CMake internally to use C++11 ...")
+    set(CMAKE_CXX_STANDARD 11)
+  endif()
+endif()
+
+# Add a macro for whether to require using std::shared_ptr.
+# depends on the version of Thrift.
+if(${THRIFT_VERSION} VERSION_LESS_EQUAL 0.10.0)
+  # old version that uses boost::shared_ptr
+  add_definitions("-DCTRLPORT_USE_STD_SHARED_PTR=0")
```

Comment:
I think, since the definition is used in public headers it needs to be propagated to OOTs, i.e., it should be a public compile definition of the runtime target.

Reply:
Good catch!